### ISSUE TITLE
Transtive maven dependencies that have exclusions defined in their POM can cause the dependency itself to be excluded

### ISSUE THREAD

**Issue by dpratt** [sbt/sbt#2770]
## steps

Create a simple project that has a Maven dependency - for example in a basic build.sbt

``` scala
 libraryDependencies += "org.elasticsearch.plugin" % "transport-netty4-client" % "5.0.0-beta1"
```

then do 

```
sbt "show fullClasspath"
```
## problem

None of the transitive dependencies from a top-level dependency that have a global wildcard exclusion in their POM are being included in the classpath in SBT.

The published POM for transport-netty4-client contains a dependency entry that looks like the following - 

``` xml
    <dependency>
      <groupId>io.netty</groupId>
      <artifactId>netty-codec</artifactId>
      <version>4.1.5.Final</version>
      <scope>compile</scope>
      <exclusions>
        <exclusion>
          <groupId>*</groupId>
          <artifactId>*</artifactId>
        </exclusion>
      </exclusions>
    </dependency>
```

My suspicion is that the POM parsing code in the ivy resolver in SBT is incorrectly causing a dependency to exclude itself if there's a wildcard in it's exclusions.
## expectation

To see how it should behave, put the following in a pom.xml somewhere

``` xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
  <modelVersion>4.0.0</modelVersion>
  <groupId>test</groupId>
  <artifactId>test</artifactId>
  <version>1.0.0-SNAPSHOT</version>
  <dependencies>
    <dependency>
      <groupId>org.elasticsearch.plugin</groupId>
      <artifactId>transport-netty4-client</artifactId>
      <version>5.0.0-beta1</version>
    </dependency>
  </dependencies>
</project>
```

and do 'mvn dependency:tree' - you'll see that it outputs a bunch of transitive dependencies that don't show up when used in SBT.

**Comment by eed3si9n** [comment]
I think this is a duplicate of #1431, which has been fixed in 0.13 branch (but hasn't been released yet in latest). Could you test sbt 0.13.13-RC2 to see if this has been fixed?

**Comment by dpratt** [comment]
You're correct - looks like this is a dupe of #1431 - updating to 0.13.13-RC2 fixes it.
